### PR TITLE
[Breaking] Refactor Tracers

### PR DESCRIPTION
**Pull request by vowelparrot**
### Refactor the BaseTracer
- Remove the 'session' abstraction from the BaseTracer
- Rename 'RunV2' object(s) to be called 'Run' objects (Rename previous Run objects to be RunV1 objects)
- Ditto for sessions: TracerSession*V2 -> TracerSession*
- Remove now deprecated conversion from v1 run objects to v2 run objects in  LangChainTracerV2
- Add conversion from v2 run objects to v1 run objects in V1 tracer

### CODE REVIEW

File: langchain/callbacks/manager.py [hunk 1]
```diff
@@ -81,11 +83,11 @@ def tracing_v2_enabled(
     )
     if isinstance(example_id, str):
         example_id = UUID(example_id)
-    cb = LangChainTracerV2(example_id=example_id)
-    session = cast(TracerSessionV2, cb.new_session(session_name))
-    tracing_callback_var.set(cb)
```
Comment (vowelparrot) [inline langchain/callbacks/manager.py:86]: Was the original intentional? Seems like it should be the v2 callback var here?

File: langchain/callbacks/tracers/__init__.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 """Tracers that record execution of LangChain runs."""
 
-from langchain.callbacks.tracers.langchain import LangChainTracer
+from langchain.callbacks.tracers.langchain_v1 import LangChainTracerV1
 
-__all__ = ["LangChainTracer"]
+__all__ = ["LangChainTracerV1"]
```
Comment (vowelparrot) [inline langchain/callbacks/tracers/__init__.py:5]: Still not exporting on the tracers/ level

File: langchain/callbacks/tracers/base.py [hunk 3]
```diff
@@ -322,3 +288,33 @@ def __deepcopy__(self, memo: dict) -> BaseTracer:
     def __copy__(self) -> BaseTracer:
         """Copy the tracer."""
         return self
+
+    def _on_llm_start(self, run: Run) -> None:
```
Comment (vowelparrot) [inline langchain/callbacks/tracers/base.py:292]: @agola11 these are what you had in mind, right?

Comment (agola11) [reply langchain/callbacks/tracers/base.py:265]: yup exactly!

File: langchain/callbacks/__init__.py [hunk 4]
```diff
@@ -3,14 +3,12 @@
 from langchain.callbacks.aim_callback import AimCallbackHandler
 from langchain.callbacks.clearml_callback import ClearMLCallbackHandler
 from langchain.callbacks.comet_ml_callback import CometCallbackHandler
-from langchain.callbacks.manager import (
-    get_openai_callback,
-    tracing_enabled,
-)
+from langchain.callbacks.manager import get_openai_callback, tracing_enabled
```
Comment (agola11) [inline langchain/callbacks/__init__.py:6]: why was this changed?

File: langchain/callbacks/manager.py (hunk 1 above)
Comment (agola11) [reply langchain/callbacks/manager.py:86]: yup this was a msitake good catch

**Review by agola11 — APPROVED** [review]
woo! nice!

File: tests/unit_tests/callbacks/tracers/test_langchain_v1.py [hunk 5]
```diff
@@ -0,0 +1,684 @@
+"""Test Tracer classes."""
```
Comment (agola11) [inline tests/unit_tests/callbacks/tracers/test_langchain_v1.py:1]: think we should split this up into two files: one with tests for basetracer (which uses faketracer) and one which checks the conversion and uses the langchain_tracer_v1 fixture

**Review by agola11 — COMMENTED** [review]
whoops actually forgot to look at one file

**Review by agola11 — COMMENTED** [review]
whoops actually forgot to look at one file

File: tests/unit_tests/callbacks/tracers/test_langchain_v1.py [hunk 6]
```diff
@@ -0,0 +1,684 @@
...
+        """Create a new tracing session."""
+        return new_session(name)
+
+    def load_session(self, session_name: str) -> TracerSessionV1:
+        """Load a tracing session."""
+        return load_session(session_name)
+
+    def load_default_session(self) -> TracerSessionV1:
+        """Load a tracing session."""
+        return load_default_session()
```
[the remark this answers is not in the document]
Comment (vowelparrot) [reply tests/unit_tests/callbacks/tracers/test_langchain_v1.py:95]: Yes!

File: tests/unit_tests/callbacks/tracers/test_langchain_v1.py (hunk 5 above)
Comment (vowelparrot) [reply tests/unit_tests/callbacks/tracers/test_langchain_v1.py:1]: Sounds good

File: langchain/callbacks/__init__.py (hunk 4 above)
Comment (vowelparrot) [reply langchain/callbacks/__init__.py:6]: Hm no idea. Must have been an IDE auto-format - I'll remove the original export though - can be a subsequent one
